tests/test_machine.py

```diff
@@ -1,69 +1,68 @@
-    pytest.mark.parametrize("verb", ["boot", "qemu"]),
-    pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")
-]
-
-
-def test_simple_run(verb: str) -> None:
-    with machine.Machine([verb]) as m:
-        p = m.run(["echo", "This is a test."])
-        assert "This is a test." == p.stdout.strip("\n")
-
-    assert m.exit_code == 0
-
+class MkosiMachineTest(unittest.TestCase):
+    reboot: bool
+    args: str
+    machine: Machine
 
-def test_wrong_command(verb: str) -> None:
-    # First tests with argument check = True from mkosi.backend.run(), therefore we see if an exception is raised
-    with machine.Machine([verb]) as m:
-        with pytest.raises(MkosiException):
-            m.run(["NonExisting", "Command"])
-        with pytest.raises(MkosiException):
-            m.run(["ls", "NullDirectory"])
+    def __init_subclass__(cls, /, args: str, reboot: bool = True) -> None:
+        cls.reboot = reboot
+        cls.args = args
 
-    assert m.exit_code == 0
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.machine = Machine([cls.args])
+        cls.machine.__enter__()
 
-    # Second group of tests with check = False to see if stderr and returncode have the expected values
-    with machine.Machine([verb]) as m:
-        result = m.run(["NonExisting", "Command"], check=False)
-        assert result.returncode in (203, 127)
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.machine.__exit__(*exc_info())
 
-        result = m.run(["ls", "-"], check=False)
-        assert result.returncode == 2
-        assert "No such file or directory" in result.stderr
+    def tearDown(self) -> None:
+        if self.reboot:
+            self.machine.reboot_image()
 
-    assert m.exit_code == 0
+class MkosiMachineTester(MkosiMachineTest, args="boot", reboot=True):
+    def test_simple_run(self) -> None:
+        process = self.machine.run(["echo", "This is a test."])
+        self.assertEqual(process.stdout.strip("\n"), "This is a test.")
 
+    def test_wrong_command(self) -> None:
+        # Check = True from mkosi.backend.run(), therefore we see if an exception is raised
+        with self.assertRaises(MkosiException):
+            self.machine.run(["NonExisting", "Command"])
+        with self.assertRaises(MkosiException):
+            self.machine.run(["ls", "NullDirectory"])
 
-def test_infinite_command(verb: str) -> None:
-    with machine.Machine([verb]) as m:
-        with pytest.raises(TimeoutExpired):
-            m.run(["tail", "-f", "/dev/null"], 2)
+        # Check = False to see if stderr and returncode have the expected values
+        result = self.machine.run(["NonExisting", "Command"], check=False)
+        self.assertTrue(result.returncode in (203, 127))
 
-    assert m.exit_code == 0
+        result = self.machine.run(["ls", "-"], check=False)
+        self.assertEqual(result.returncode, 2)
+        self.assertTrue("No such file or directory" in result.stderr)
```

Comment:
We'll still be using pytest as the test runner so we can just use `assert` here instead of the unittest specific methods